martin/src/bin/martin-cp.rs

```diff
@@ -442,6 +441,47 @@ async fn main() {
     }
 }
 
+fn setup_logging() {
+    use tracing_subscriber::filter::EnvFilter;
+    use tracing_subscriber::fmt::Layer;
+    use tracing_subscriber::prelude::*;
+    // transform log records into `tracing` `Event`s.
+    tracing_log::LogTracer::builder()
+        .init()
+        .expect("the global logger to only be set once");
+
+    let log_format = LogFormat::from_env();
+    let registry = tracing_subscriber::registry()
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(tracing::Level::INFO.into())
+                .from_env_lossy(),
+        )
+        .with((log_format == LogFormat::Compact).then(|| Layer::default().compact()))
```

Comment:
oh come on, this should be a clean `match` statement  :)

Reply:
That would work only if this would not so in depth generics.
If I were to match on this, I would likely end up with a dyn-box, duplicate the usage below or similar.

Imo, the current version is simpler..